File: zip_extension.py
import zipfile
import os

def zipdir(path, ziph):
    for root, dirs, files in os.walk(path):
        for file in files:
            ziph.write(os.path.join(root, file), 
                       os.path.relpath(os.path.join(root, file), path))

if __name__ == '__main__':
    zipf = zipfile.ZipFile('public/LinkedInExtension.zip', 'w', zipfile.ZIP_DEFLATED)
    zipdir('extension/', zipf)
    zipf.close()
    print("Zipped successfully")

File: zip_extension.js
const fs = require('fs');
const archiver = require('archiver');
const path = require('path');

const output = fs.createWriteStream(path.join(__dirname, 'public', 'LinkedInExtension.zip'));
const archive = archiver('zip', { zlib: { level: 9 } });

output.on('close', function() {
  console.log(archive.pointer() + ' total bytes appended');
  console.log('Archived successfully.');
});

archive.on('error', function(err) {
  throw err;
});

archive.pipe(output);
archive.directory('extension/', false);
archive.finalize();
